flexget/plugins/list/ombi_list.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import unicode_literals, division, absolute_import
+from builtins import *  # noqa pylint: disable=unused-import, redefined-builtin
+from future.moves.urllib.parse import urlparse
+
+import logging
+from collections import MutableSet
+
+import requests
+from requests import RequestException
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+
+log = logging.getLogger('ombi_list')
+
+class OmbiBase(object):
+
+    @staticmethod
+    def generate_title(item, strip):
+        if item.get('releaseDate') and not strip:
+            return '%s (%s)' % (item.get('title'), item.get('releaseDate')[0:4] )
+        else:
+            return item.get('title')
+
+    @staticmethod
+    def construct_url(config):
+        parsedurl = urlparse(config.get('base_url'))
+        if config.get('type') in ['movies']:
+            log.debug('Received movie list request')
+            return '%s://%s:%s%s/api/v1/Request/movie?apikey=%s' % (parsedurl.scheme, parsedurl.netloc, config.get('port'), parsedurl.path, config.get('api_key'))
+        elif config.get('type') in ['shows', 'seasons', 'episodes']:
+            log.debug('Received TV list request')
+            return '%s://%s:%s%s/api/v1/Request/tv?apikey=%s' % (parsedurl.scheme, parsedurl.netloc, config.get('port'), parsedurl.path, config.get('api_key'))
+        else:
+            raise plugin.PluginError('Error: Unknown list type %s.' % (config.get('type')))
+
+    @staticmethod
+    def get_json(url):
+        try:
+            return requests.get(url).json()
+        except RequestException as e:
+            raise plugin.PluginError('Unable to connect to Ombi at %s. Error: %s' % (url, e))
+
+    @staticmethod
+    def generate_entry(config, parent_request, child_request=False, season=False, episode=False):
+        log.debug('Generating Entry')
+        entry = Entry()
+        if config.get('type') == 'movies':
```

Comment:
This function does ONE thing, it builds an entry for the list, each type has fairly different fields, passing the config and using the type is very convenient. I see no need for additional functions.

Reply:
It's convenient, but you very quickly end up with a mess. I already think it's messy. It may be convenient, but it's easier to read if you create separate functions eg. `generate_episode_entry`, where you pass the proper arguments. Just by looking at the function and its arguments, you can easily see what it does without reading through the code.

This is a prime example of the mess this approach can create: https://github.com/Flexget/Flexget/blob/develop/flexget/plugins/internal/api_trakt.py#L859-L1105